plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts

```diff
@@ -64,4 +71,79 @@ export class CatalogIdentityClient {
 
     return items[0] as UserEntity;
   }
+
+  /**
+   * Resolve additional entity claims from the catalog, using the passed-in subject and entity
+   * claims. Designed to be used within a `signInResolver` where additional entity claims might be
+   * provided, but group membership and transient group membership lean on imported catalog
+   * relations.
+   *
+   * Returns a claim structure that can be passed directly to `issueToken`, with the same sub and a
+   * superset of `ent` claims.
+   */
+  async resolveCatalogMemberClaims(
```

Comment:
I wonder if we can make this an object and destructure so we can have named arguments and evolve this API easier?

Reply:
And could we detach it from the intricacies of token shapes? We could make it be about "return all the things that can be reached through membership in the catalog, given a set of roots". It doesn't even change the shape of `sub` at the moment. This would male the function more general at little cost.

```ts
const sub = 'freben';
const ent = ['user:github/freben'];
const fullEnt = extendMembershipGraph([sub].concat(ent));
```